FutureNorthants/northants-website
# Backend support for Promotions slice

### Setup
Checkout this branch and run a `./drush.wrapper @docker config:import` and `cache:rebuild`

### Testing
1. Verify that you can now insert a promotions slice in the body of any page and add up to 4 promotional tiles within the slice. TODO: need a slice icon from Lawrence.
2. Go to https://core-cms.northants.localhost/graphql and enter the following query:
```graphql
{
  route(path: "your-page-route") {
    node {
      title
      body {
        value
        embeds {
          id
          paragraph {
            ... on PromotionsParagraph {
              promotions {
                id
                imageLarge
                imageMedium
                imageSmall
                imageAltText
                title
                body {
                  value
                }
                link {
                  title
                  url
                }
              }
            }
          }
        }
      }
    }
  }
}
```
...and verify it includes the data you expect.
